libs/ballot-interpreter/src/legacy_adapter.ts

```diff
@@ -407,13 +408,29 @@ function convertNextInterpretedBallotPage(
 /**
  * Converts the result of the NH interpreter to the legacy interpreter result format.
  */
-export function convertNhInterpretResultToLegacyResult(
+export async function convertNhInterpretResultToLegacyResult(
   options: InterpreterOptions,
-  nextResult: NextInterpretResult
-): InterpretResult {
+  nextResult: NextInterpretResult,
+  sheet: SheetOf<string>
+): Promise<InterpretResult> {
   /* istanbul ignore next */
   if (nextResult.isErr()) {
-    return err(new Error(JSON.stringify(nextResult.err())));
+    return ok([
+      {
+        interpretation: {
+          type: 'UnreadablePage',
+          reason: JSON.stringify(nextResult.err().type),
```

Comment:
Was there a particular reason to `JSON.stringify` this? The `type` is just a string, so we could just pass it through as-is.

Reply:
nope, copy-pasted and then changed the content and forgot to remove the stringify. I'll update.